src/server/index.js

```diff
@@ -109,96 +115,219 @@ const createStock = quote => {
   };
 };
 
-const calculateDateYearsInPast = years => {
-  const year = new Date().getFullYear() - years;
-  const date = new Date();
-  date.setFullYear(year);
-  return date;
+// Used for historical() data obtained using period 'd'
+const getDatesAndPrices = dailyData => {
+  let datesAndPrices = [];
+  dailyData.forEach(({
+    date,
+    close
+  }) => {
+    datesAndPrices.unshift({
+      date: formatDateForMaxStockData(date),
+      price: close
+    });
+  });
+  return datesAndPrices;
+};
+
+// NOTE: this date has timezone UTC, which is incorrect but works in this
+// case because we are just extracting the time
+const getAdjustedDateForTimestamp = (gmtoffset, timestamp) => {
+  const adjustedTimestamp =
+    (timestamp + gmtoffset) * constants.MILLISECONDS_PER_SECOND;
+  return new Date(adjustedTimestamp);
 };
 
-const padSingleDigitWithZero = value => {
-  let num = parseInt(value);
-  // need to check value because parseInt turns '12hello' into a number
-  if (isNaN(value) || isNaN(num)) {
-    throw new TypeError(`${padSingleDigitWithZero.name} requires a number or numeric string`);
+const getDateAndTime = (gmtoffset, timestamp, dateAndTimeFormat) => {
+  const dateAndTime = getAdjustedDateForTimestamp(gmtoffset, timestamp);
+  return dateFormat(dateAndTime, dateAndTimeFormat, true);
+};
+
+const getStartOfDayTimestampIndex = (dayIndex, timestampsPerDay) =>
+  Math.floor(dayIndex * timestampsPerDay);
+
+const getEndOfDayTimestampIndex = (dayIndex, timestampsPerDay) =>
+  Math.floor((dayIndex + 1) * timestampsPerDay);
+
+// Could be buggy depending on how the timestamps in their API are given.
+// Couldn't find info on this so will need to test at certain times of day.
```

Comment:
did you do this? delete this line if so

Reply:
No, because it's the weekend so nothing is changing. Leaving it in until it's thoroughly tested but I have tested it repeatedly since I wrote it and it's always worked